assembly/src/errors.rs

```diff
@@ -108,24 +98,6 @@ pub enum AssemblyError {
     },
     #[error(transparent)]
     #[diagnostic(transparent)]
-    Parsing(#[from] crate::parser::ParsingError),
```

Comment:
These are all able to be removed because we use `Report` in most places now. I think we could remove `AssemblyError` entirely simply by switching the remaining instances where we construct the remaining variants to use the `miden_assembly::diagnostics::miette::diagnostic!` macro instead.

I think I'd probably improve the ergonomics one step further over `diagnostic!` though, and define our own version of `miette!` (which is basically `diagnostic!` wrapped in a `Report`), and export _that_ as `diagnostic!`. Currently though, we aren't exporting any of those macros from `miden_assembly::diagnostics`, so that would be needed. The end result would be a way richer and more ergonomic way of raising diagnostics versus types like `AssemblyError`.

Reply:
Yep - I'm fine with removing `AssemblyError` - but that's for another PR.